### DISCUSSION TITLE
how to make the "Error loading webview: Service Workers are not enabled in browser. Webviews will not work."  not show again

### DISCUSSION THREAD

**Discussion by shangguandong1996** [coder/code-server#2821] [Q&A] [answered]
Hi, every one
when I plot some thing in Vscode each time, there are always appear the `Error loading webview: Service Workers are not enabled in browser. Webviews will not work.` 
![image](https://user-images.githubusercontent.com/22555126/110093319-8c488100-7dd5-11eb-8839-1b29e5cf5c78.png)

I have seen some issue about it, but I do not want to fix it, I just want this error do not appear again.
Can someone give me some adivce?

Best wishes
Guandong Shang

**Comment by code-asher** [comment, marked as answer]
The only way to prevent the error from appearing is to fix the error. It's most likely caused by using an insecure context so using HTTPS would fix it (even a self-signed certificate will work which code-server can generate for you).

**Reply by shangguandong1996** [reply]
thanks, I will try to fix it :)

**Reply by shangguandong1996** [reply]
But I find a werid things that it seems **Firefox** will not show the error again when I set a self-signed certificate while **chrome** still show……

**Reply by code-asher** [reply]
Oh hmmm yeah it looks like Chrome now requires a valid certificate before it will enable service workers. That's unfortunate. I suppose that means the only way to fix Chrome is to get a valid certificate (from Let's Encrypt for example) or use `--link` to proxy through our cloud (in which case we provide the certificate).
